api/src/main/java/org/pmiops/workbench/db/dao/UserService.java

```diff
@@ -100,11 +100,30 @@ private User updateWithRetries(Function<User, User> userModifier, User user) {
   }
 
   private void updateDataAccessLevel(User user) {
-    boolean shouldBeRegistered = Optional.ofNullable(user.getIdVerificationIsValid()).orElse(false)
-        && user.getDemographicSurveyCompletionTime() != null
-        && user.getTrainingCompletionTime() != null
-        && user.getTermsOfServiceCompletionTime() != null
+    // TODO: Add in when we add this module
+    // boolean dataUseAgreementCompliant = user.getDataUseAgreementCompletionTime() != null ||
+    // user.getDataUseAgreementBypassTime() != null || !configProvider.get().access.enableDataUseAgreement;
+    Timestamp current = new Timestamp(clock.instant().toEpochMilli());
+    boolean complianceTrainingNotExpired = user.getComplianceTrainingExpirationTime() == null ||
+      current.before(user.getComplianceTrainingExpirationTime()) ||
+      !(configProvider.get().access.enableComplianceTraining);
+    boolean complianceTrainingCompliant = complianceTrainingNotExpired &&
+      (user.getComplianceTrainingCompletionTime() != null ||
+      user.getComplianceTrainingBypassTime() != null ||
+      !(configProvider.get().access.enableComplianceTraining));
+    // TODO: add in the check on completion time once RW-2235 (Scout's) has been merged in
+    boolean eraCommonsCompliant = user.getEraCommonsBypassTime() != null || !configProvider.get().access.enableEraCommons;
+    boolean idVerificationCompliant = user.getIdVerificationCompletionTime() != null ||
+      user.getIdVerificationBypassTime() != null || !configProvider.get().access.enableIdVerification ||
+      Optional.ofNullable(user.getIdVerificationIsValid()).orElse(false);
```

Comment:
It's worth making a note here that we need to eventually migrate from the "id_verification_is_valid" field to the "id_verification_completion_time" field... maybe a TODO above this line of logic?